src/coreclr/scripts/superpmi.py

```diff
@@ -2259,6 +2271,48 @@ def filter_superpmi_collections(path):
     print("")
 
 
+def merge_mch(coreclr_args):
+    """ Merge all the MCH files in a directory into a single output MCH file.
+        This is a utility function mostly for use by the CI scripting. It is a
+        thin wrapper around:
+
+            mcs -merge <output_mch_path> <mch_dir> -recursive -dedup -thin
+            mcs -toc <output_mch_path>
+
+        Note that we expect <mch_dir> to be a directory tree that contains only
+        MC or MCH files, and all of them will be merged. This is what "mcs -merge"
+        expects. If we need more flexibility, we can either add more flexibility
+        to "mcs -merge" (preferable), or work around it here, e.g., by creating
+        a temp directory and copying everything we want merged to it before invoking
+        the merge command.
+
+    Args:
+        coreclr_args (CoreclrArguments) : parsed args
+
+    Returns:
+        True on success, else False
+    """
+
+    mcs_path = determine_mcs_tool_path(coreclr_args)
+    command = [mcs_path, "-merge", coreclr_args.output_mch_path, coreclr_args.mch_dir, "-recursive", "-dedup", "-thin"]
```

Comment:
I have a small request. Since `mcs.exe` takes `pattern` parameter (`-merge outputfile pattern [-dedup [-thin]]` from help docs), is it possible to add a parameter to `merge-mch` such that it only scans for files of particular pattern inside `mch_dir`? The reason for this ask is, in CI, I will have bunch of `libraries*.mch` files and `test*.mch` files and I want to merge them separately producing `libraries.mch` and `test.mch`.

Reply:
I can just take "pattern" instead of "mch_dir", and just pass it through to mcs, and expect you to pass `-pattern c:\my\dir\*.mch`

Reply:
that should work. I am hoping it will work recursively like `c:\my\dir\*\libraries*.mch`

Reply:
`mcs` doesn't handle any wildcards except in the filename pattern at the end.

Reply:
In that case, i think we will still need `-mch_dir` so I can do  
- `superpmi.py mcs-merge -mch_dir c:\my\dir -pattern libraries*.mch -output_mch_file libraries.mch` 
- `superpmi.py mcs-merge -mch_dir c:\my\dir -pattern test*.mch -output_mch_file test.mch`

Reply:
Can't you just do:
```
superpmi.py mcs-merge -pattern c:\my\dir\libraries*.mch -output_mch_file libraries.mch
superpmi.py mcs-merge -pattern c:\my\dir\test*.mch -output_mch_file test.mch
```

Reply:
It's going to use `mcs -merge -recursive` so it will recurse to all directories in the `c:\my\dir` tree.

Reply:
No, the problem is helix files are uploaded based on workitems.

![image](https://user-images.githubusercontent.com/12488060/94197474-b0895480-fe6a-11ea-8933-7781cc3cf603.png)